angular-notification.js

```diff
@@ -121,18 +123,21 @@ function NotificationProvider() {
 
     /**
      * Static method to request permission.
-     *
-     * @param {Function} callback
+     * It returns a promise
      */
 
-    NgNotification.requestPermission = function (callback) {
-      if (! $window.Notification) return false;
+    NgNotification.requestPermission = function () {
+        var deferred = $q.defer();
```

Comment:
@davinkevin I think it's preferable to use the promise like implementation than deffered:

```js
return $q(function (resolve, reject) { ... });
```